app/domain/authentication/authn_k8s/consts.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+module Authentication
+  module AuthnK8s
+
+    K8S_AUTHENTICATOR_NAME = 'authn-k8s'
+
+    module Restrictions
+
+      NAMESPACE = "namespace"
+      SERVICE_ACCOUNT = "service-account"
+      POD = "pod"
+      DEPLOYMENT = "deployment"
+      STATEFUL_SET = "stateful-set"
+      DEPLOYMENT_CONFIG = "deployment-config"
+      AUTHENTICATION_CONTAINER_NAME = "authentication-container-name"
```

Comment:
this annotation is not really a restriction iirc. It just tells conjur to which container it should inject the cert. 

To verify this - what if the host has all the annotations correct, but has this value set to the application container in the pod. i _think_ that the authentication will pass (but of course the cert will be injected to the wrong container).

am i right?

Reply:
Partially.
The authentication will pass, because the restriction stands.
We have a dedicated step for validating the container name: https://github.com/cyberark/conjur/blob/master/app/domain/authentication/authn_k8s/validate_pod_request.rb#L34
This step looks for a container with the required name in the pod, and throws a `ContainerNotFound` if it doesn't find any.
This is a restriction validation, just like any other.

Reply:
i won't push any further than this but it is not really `a restriction validation, just like any other`. We do not enforce the request to come from the container mentioned in the annotation, as we do with the rest. We will validate that the container exists in the pod but it can be any other container. 

Having said that, if you think it makes the code more readable to treat this annotation as a resource restriction i am ok with that. it may be worth mentioning in a comment that it doesn't act exactly as a resource restriction.

Reply:
or maybe just move it outside of restrictions in this class to identify that.

Reply:
done